### ISSUE TITLE
A proposal on keymaps management

### ISSUE THREAD

**Issue by shofel** [shaunsingh/nyoom.nvim#31]
Tldr: What if we remove custom wrappers around which-key and use it directly everywhere to set and describe key mappings? It should reduce complexity of both maintenance and usage.

## A longer version

1. Let's use which-key to not only document keymap roots, but also to map all the keys.
2. Let's use which-key directly, without wrappers. Which-key has a neat declarative interface. If it has some rough edges, we can probably fix them with small specialized macros. But first of all, those issues should be carefully identified.
3. Describe key mappings for lsp in the same file as other mappings. Then [export a function](https://github.com/shofel/nyoom.nvim/blob/831f1061b5b53bd07cc4bf662f7702bce89f8728/fnl/core/keymaps.fnl#L72) to setup keys for a buffer, and use this function in the lsp configuration

How does it looks like:
https://github.com/shofel/nyoom.nvim/blob/831f1061b5b53bd07cc4bf662f7702bce89f8728/fnl/core/keymaps.fln

## Profit of removing `macros.keybind-macros`
1. Reduce maintenance cost
2. Reduce learning efforts

**Comment by shaunsingh** [comment]
I agree with you on this. I was trying to make some macros for which-key usage (so I could make it similar to doom-emacs's syntax: https://github.com/doomemacs/doomemacs/blob/3b789752a30419535b328c605957b6b43a9d47fa/modules/lang/emacs-lisp/demos.org#map) but it seems likes its more trouble than its worth. I'll rewrite the "core" bindings using which-key for now.

**Comment by shaunsingh** [comment]
Keymaps are now set using which-key! Thanks for the suggestion
